lib/telephony/pinpoint/sms_sender.rb

```diff
@@ -13,43 +15,65 @@ class SmsSender
 
       # rubocop:disable Metrics/MethodLength
       def send(message:, to:)
-        @pinpoint_response = pinpoint_client.send_messages(
-          application_id: Telephony.config.pinpoint.sms.application_id,
-          message_request: {
-            addresses: {
-              to => {
-                channel_type: 'SMS',
+        last_response = nil
+        client_configs.each do |client_config|
+          pinpoint_response = client_config.client.send_messages(
+            application_id: client_config.config.application_id,
+            message_request: {
+              addresses: {
+                to => {
+                  channel_type: 'SMS',
+                },
               },
-            },
-            message_configuration: {
-              sms_message: {
-                body: message,
-                message_type: 'TRANSACTIONAL',
-                origination_number: Telephony.config.pinpoint.sms.shortcode,
+              message_configuration: {
+                sms_message: {
+                  body: message,
+                  message_type: 'TRANSACTIONAL',
+                  origination_number: client_config.config.shortcode,
+                },
               },
             },
-          },
-        )
-        response
+          )
+
+          response = build_response(pinpoint_response)
+          return response if response.success?
+          notify_pinpoint_failover(response.error)
+          last_response = response
+        end
+        last_response
       end
       # rubocop:enable Metrics/MethodLength
 
-      private
+      # @api private
+      # An array of (client, config) pairs
+      # @return [Array<ClientConfig>]
+      def client_configs
+        @client_configs ||= Telephony.config.pinpoint.sms_configs.map do |sms_config|
+          credentials = AwsCredentialBuilder.new(sms_config).call
+          args = { region: sms_config.region, retry_limit: 1 }
```

Comment:
Seeing the retry_limit param here reminds me of a caveat.

IIRC the request here has a 5 second timeout. With 1 retry that means each config has 10 seconds total is can spend sending an SMS. If Pinpoint is timing out that means 2 configs will take 20 seconds. That will push the user's request to the IdP outside the 15 second timeout window.

Since we have a failover, does it makes sense to go to no retries?

Reply:
Yeah happy to remove that.

Also that reminded me I was talking to @pauldoomgov about setting a timeout that covers our 95th percentile, it I think it's much less than 5 seconds

Reply:
Funny thing - NewRelic seems to link all we need is an average :(   So I am not sure what our p95 is on PinPoint calls.  I would guess `1.5s` would be a sane starting point.   That would give us 12sec if we fail, retry, failover+fail, retry.   I think @jmhooper has a point about not retrying on failover. 

One other item: Might be good to add timing output to the telephony output.

Reply:
For reference - 3hr window of PinPoint API calls in Prod

![Screen Shot 2020-05-27 at 14 49 24](https://user-images.githubusercontent.com/59626817/83065567-52b63e80-a029-11ea-9ad7-85d5f8ccabd2.png)